packages/nox/src-loca/src/LOCA_Stepper.C

```diff
@@ -588,7 +589,16 @@ LOCA::Stepper::compute(LOCA::Abstract::Iterator::StepStatus stepStatus)
   printStartStep();
 
   // Compute next point on continuation curve
-  solverStatus = solverPtr->solve();
+  try
+  {
+    solverStatus = solverPtr->solve();
+  }
+  catch (const NOX::Exceptions::SolverFailure& e)
+  {
+    std::cerr << "Caught NOX::Exceptions::SolverFailure:" << std::endl
```

Comment:
This will print on all MPI processes.  Is there a way to restrict this to print on only one MPI process?  If I remember right, NOX has an abstraction for that.

Reply:
Good catch @mhoemmen,

@jmgate, all outputtting should be using a Teuchos::FancyOStream.  It is best to use the one gotten from the VerboseObject interface which every Thyra object derives from:

* https://trilinos.org/docs/dev/packages/teuchos/doc/html/classTeuchos_1_1VerboseObjectBase.html#ade4feddc2032e48c6ffe13be0b5e513d

That will ensure the correct outputting is done.

Also note that it is generally a good idea to print the string returned by `e.what()`.  That often contains some useful info for the user.

Reply:
NOX output was written before FancyOStream existed. It has no dependency on FancyOStream. To be consistent with the rest of the nox package, use the print utils object. It has accessors to different ostreams that can be set by the user. They combine an output or error stream with either serial (one process prints) or parallel (all processes print). 

https://trilinos.org/docs/dev/packages/nox/doc/html/classNOX_1_1Utils.html

```

std::ostream & | out () const
  | Returns the std::ostream for printing if this proces is the print process. Returns a Teuchos::oblackholestream otherwise.
 
std::ostream & | out (NOX::Utils::MsgType type) const
  | Returns the std::ostream for printing if this process is the print process and the print type is valid. Returns a Teuchos::oblackholestream otherwise.
 
std::ostream & | pout () const
  | Returns  the std::ostream for printing regardless of the print processor. Only  use this call if you want all processes to print to the std::ostream.
 
std::ostream & | pout (NOX::Utils::MsgType type) const
  | Returns  the std::ostream for printing if the print type matches. Only use this  call if you want all processes to print to the std::ostream for the  print type.
 
std::ostream & | err () const
  | Returns the error stream for printing if this is the print process.
 
std::ostream & | perr () const
  | Returns  the error stream for printing to all processors. Only use this call if  you want all processes to print to the error stream.
```

Reply:
Testing this change now.  For future reference, in case anyone wants to go fix it (I officially un-volunteer myself), before I committed these changes I `grep`ed through NOX for `std::cerr`, found it written to directly all over the place, assumed that was an acceptable practice, and then did the same thing myself.